packages/react/src/components/Authenticator/SelectMfa/SelectMfa.tsx

```diff
@@ -0,0 +1,178 @@
+import React from 'react';
+import { authenticatorTextUtil, MfaType } from '@aws-amplify/ui';
+
+import { Flex } from '../../../primitives/Flex';
+import { Heading } from '../../../primitives/Heading';
+import { Radio } from '../../../primitives/Radio';
+import { RadioGroupField } from '../../../primitives/RadioGroupField';
+import { useAuthenticator } from '@aws-amplify/ui-react-core';
+import { useCustomComponents } from '../hooks/useCustomComponents';
+import { useFormHandlers } from '../hooks/useFormHandlers';
+// import { FormFields } from '../shared/FormFields';
+import { ConfirmSignInFooter } from '../shared/ConfirmSignInFooter';
+import { RemoteErrorMessage } from '../shared/RemoteErrorMessage';
+import { RouteContainer, RouteProps } from '../RouteContainer';
+import { toLower } from 'lodash';
+
+const { getChallengeText } = authenticatorTextUtil;
+
+function parseMfaTypes(types?: MfaType[]): string {
+  if(types) {
+    return types.join();
+  } else {
+    return 'types was undefined.';
+  }
+}
+
+/*
+interface MfaOptions {
+  email: string;
+  username: string;
+}
+
+const opt: MfaOptions = {
+  email: 'EMAIL',
+  username: 'Totp Prop',
+};
+
+type MfaType = 'EMAIL' | 'wow';
+*/
+
+const generateRadioGroup = (
+  attributes?: MfaType[]
+): JSX.Element[] => {
+  const elements: JSX.Element[] = [];
+  console.log('print attributes: ', attributes);
+  attributes.forEach((value, index) => {
+    console.log('print value and index: ', value, index);
+    elements.push((
+      <Radio
+        name="mfa_selection"
+        value={value}
+        key={value}
+      >
+        Option {index}: {value}
+      </Radio>
+    ));
+  });
+  return elements;
+  /*
+  return attributes.map
+    ([key, value]: [string, string], index) => {
+      const MfaType = 
+      return (
+        <Radio
+          name="MfaAtt"
+          value={key}
+          key={key}
+          defaultChecked={index === 0}
+        >
+          {MfaType}:{' '}
+          {value}
+        </Radio>
+      );
+    }
+  );
+  */
+};
+
+
+export const SelectMfa = ({
```

Comment:
How are instances of a user only having one MFA option handled?

Reply:
The JS library should route directly to the confirm sign in step associated with that MFA option. I will make sure that behavior is tested as well.

Reply:
Perfect, thanks for confirming that that case is already handled